evm/vm/base.py

```diff
@@ -188,6 +272,8 @@ def get_block_class(self):
 
     def get_block_by_hash(self, block_hash):
         block_header = self.evm.get_block_header_by_hash(block_hash)
+        if block_header is None:
+            return None
```

Comment:
I chose to return None when the block is not found because I assumed it'd be a common idiom to use this method to check if a block has been received by the node yet, and that a block not being found is not necessarily an exceptional occurrence. Happy to change it if you disagree